src/modules/battery_status/battery_status.cpp

```diff
@@ -194,63 +189,69 @@ BatteryStatus::adc_poll()
 
 	int selected_source = -1;
 
-	if (ret >= (int)sizeof(buf_adc[0])) {
-
-		/* Read add channels we got */
-		for (unsigned i = 0; i < ret / sizeof(buf_adc[0]); i++) {
-			{
-				for (int b = 0; b < BOARD_NUMBER_BRICKS; b++) {
-
-					/* Once we have subscriptions, Do this once for the lowest (highest priority
-					 * supply on power controller) that is valid.
-					 */
-					if (selected_source < 0 && _analogBatteries[b]->is_valid()) {
-						/* Indicate the lowest brick (highest priority supply on power controller)
-						 * that is valid as the one that is the selected source for the
-						 * VDD_5V_IN
-						 */
-						selected_source = b;
+	if (_adc_report_sub.update(&adc_report)) {
+
+		for (int b = 0; b < BOARD_NUMBER_BRICKS; b++) {
+
+			/* Once we have subscriptions, Do this once for the lowest (highest priority
+			 * supply on power controller) that is valid.
+			 */
+			if (selected_source < 0 && _analogBatteries[b]->is_valid()) {
+				/* Indicate the lowest brick (highest priority supply on power controller)
+				 * that is valid as the one that is the selected source for the
+				 * VDD_5V_IN
+				 */
+				selected_source = b;
 #  if BOARD_NUMBER_BRICKS > 1
 
-						/* Move the selected_source to instance 0 */
-						if (_battery_pub_intance0ndx != selected_source) {
-							_analogBatteries[_battery_pub_intance0ndx]->swapUorbAdvert(
-								*_analogBatteries[selected_source]
-							);
-							_battery_pub_intance0ndx = selected_source;
-						}
+				/* Move the selected_source to instance 0 */
+				if (_battery_pub_intance0ndx != selected_source) {
+					_analogBatteries[_battery_pub_intance0ndx]->swapUorbAdvert(
+						*_analogBatteries[selected_source]
+					);
+					_battery_pub_intance0ndx = selected_source;
+				}
 
 #  endif /* BOARD_NUMBER_BRICKS > 1 */
-					}
+			}
 
-					/* look for specific channels and process the raw voltage to measurement data */
-					if (_analogBatteries[b]->get_voltage_channel() == buf_adc[i].am_channel) {
-						/* Voltage in volts */
-						bat_voltage_adc_readings[b] = buf_adc[i].am_data;
+			/* look for specific channels and process the raw voltage to measurement data */
+			if (adc_report.channel_id[_analogBatteries[b]->get_voltage_channel()] == _analogBatteries[b]->get_voltage_channel()) {
```

Comment:
`_analogBatteries[b]->get_voltage_channel()` is the ADC channel, which is not necessarily the index of `channel_id`.